File: Java/Array/ReverseanArray.java
import java.util.Scanner;

/**
 * ReverseanArray
 */
public class ReverseanArray {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Input array size
        System.out.print("Enter the size of the array: ");
        int size = scanner.nextInt();

        // Input array elements
        int[] arr = new int[size];
        System.out.println("Enter the elements of the array:");
        for (int i = 0; i < size; i++) {
            arr[i] = scanner.nextInt();
        }

        System.out.println("Original Array:");
        printArray(arr);

        reverseArray(arr);

        System.out.println("Reversed Array:");
        printArray(arr);

        scanner.close();
    }

    public static void reverseArray(int[] arr) {
        int start = 0;
        int end = arr.length - 1;
        while (start < end) {
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
    }

    public static void printArray(int[] arr) {
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}

File: JavaScript/Array/ReverseanArray.js
// Function to reverse an array
function reverseArray(arr) {
    let start = 0;
    let end = arr.length - 1;
    while (start < end) {
        // Swap elements at start and end indices
        let temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;
        start++;
        end--;
    }
}

// Function to print an array
function printArray(arr) {
    let output = "";
    for (let i = 0; i < arr.length; i++) {
        output += arr[i] + " ";
    }
    console.log(output);
}

// Main function
function main() {
    // Sample array
    let arr = [1, 2, 3, 4, 5];

    console.log("Original Array:");
    printArray(arr);

    reverseArray(arr);

    console.log("Reversed Array:");
    printArray(arr);
}

// Calling the main function to start the program
main();
